### ISSUE TITLE
Issue with query_objectid_as_string on Eve>=1.0.1

### ISSUE THREAD

**Issue by micaelmalta** [pyeve/eve#1468]
While migrating from Eve 1.0.0 to 1.0.1, we are seeing unexpected result with query_objectid_as_string and versioning

### Expected Behavior

if query_objectid_as_string is set to true, mongo queries should not be cast as ObjectID
```
{ find: "string_id_accounts", filter: { _id: 'xxx' }
```
### Actual Behavior

if query_objectid_as_string is set to true, mongo queries are cast as ObjectID
```
{ find: "string_id_accounts", filter: { _id: ObjectId('xxx') }
```
### Environment

* Python version: >=3.6
* Eve version: >= 1.0.1

### POC

https://github.com/micaelmalta/poc_eve

**Comment by nicolaiarocci** [comment]
@Kludex are you willing to triage and eventually give a shot at this?

**Comment by Kludex** [comment]
Yes. I'm willing to work on everything that can help us get the next release. :)

**Comment by Kludex** [comment]
The commit that introduced this behavior was: https://github.com/pyeve/eve/commit/b8d8fcd5a6d2093ec135a021f5b0f48d2c419dfe

On it.

**Comment by nicolaiarocci** [comment]
@Kludex any progress on this?

**Comment by Kludex** [comment]
No 😞

**Comment by micaelmalta** [comment]
@nicolaiarocci  is it fixed in Eve 2.0?
